// @instantdb/react-native/dist/module/Storage.native.js
import AsyncStorage from '@react-native-async-storage/async-storage';
import { StoreInterface } from '@instantdb/core';
const version = 5;
export default class Store extends StoreInterface {
    appId;
    dbName;
    constructor(appId, dbName) {
        super(appId, dbName);
        this.appId = appId;
        this.dbName = dbName;
    }
    makeKey(k) {
        return `instant_${this.appId}_${version}_${this.dbName}_${k}`;
    }
    async getItem(k) {
        const res = await AsyncStorage.getItem(this.makeKey(k));
        if (res === null)
            return res;
        return JSON.parse(res);
    }
    async setItem(k, v) {
        await AsyncStorage.setItem(this.makeKey(k), JSON.stringify(v));
    }
    async removeItem(k) {
        await AsyncStorage.removeItem(this.makeKey(k));
    }
    async getAllKeys() {
        const allKeys = await AsyncStorage.getAllKeys();
        const keys = [];
        const keyPrefix = this.makeKey('');
        for (const key of allKeys) {
            if (key.startsWith(keyPrefix)) {
                keys.push(key.substring(keyPrefix.length));
            }
        }
        return keys;
    }
    async multiSet(keyValuePairs) {
        await AsyncStorage.multiSet(keyValuePairs.map(([k, v]) => [this.makeKey(k), JSON.stringify(v)]));
    }
}
