// @instantdb/react-native/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.StoreInterface = exports.SyncTableCallbackEventType = exports.setInstantWarningsEnabled = exports.InstantAPIError = exports.InstantReactNativeDatabase = exports.createInstantRouteHandler = exports.i = exports.lookup = exports.tx = exports.id = exports.init_experimental = void 0;
exports.init = init;
require("react-native-get-random-values");
const Storage_1 = __importDefault(require("./Storage"));
const EventSourceImpl_1 = __importDefault(require("./EventSourceImpl"));
const NetworkListener_1 = __importDefault(require("./NetworkListener"));
const version_1 = __importDefault(require("./version"));
const Streams_1 = require("./Streams");
const react_common_1 = require("@instantdb/react-common");
const core_1 = require("@instantdb/core");
Object.defineProperty(exports, "i", { enumerable: true, get: function () { return core_1.i; } });
Object.defineProperty(exports, "id", { enumerable: true, get: function () { return core_1.id; } });
Object.defineProperty(exports, "tx", { enumerable: true, get: function () { return core_1.tx; } });
Object.defineProperty(exports, "lookup", { enumerable: true, get: function () { return core_1.lookup; } });
Object.defineProperty(exports, "InstantAPIError", { enumerable: true, get: function () { return 
    // error
    core_1.InstantAPIError; } });
Object.defineProperty(exports, "setInstantWarningsEnabled", { enumerable: true, get: function () { return core_1.setInstantWarningsEnabled; } });
Object.defineProperty(exports, "SyncTableCallbackEventType", { enumerable: true, get: function () { return 
    // sync table enums
    core_1.SyncTableCallbackEventType; } });
Object.defineProperty(exports, "createInstantRouteHandler", { enumerable: true, get: function () { return 
    // types
    core_1.createInstantRouteHandler; } });
Object.defineProperty(exports, "StoreInterface", { enumerable: true, get: function () { return core_1.StoreInterface; } });
/**
 *
 * The first step: init your application!
 *
 * Visit https://instantdb.com/dash to get your `appId` :)
 *
 * @example
 *  import { init } from "@instantdb/react-native"
 *
 *  const db = init({ appId: "my-app-id" })
 *
 *  // You can also provide a schema for type safety and editor autocomplete!
 *
 *  import { init } from "@instantdb/react-native"
 *  import schema from ""../instant.schema.ts";
 *
 *  const db = init({ appId: "my-app-id", schema })
 *
 *  // To learn more: https://instantdb.com/docs/modeling-data
 */
function init(
// Allows config with missing `useDateObjects`, but keeps `UseDates`
// as a non-nullable in the InstantConfig type.
config) {
    const configStrict = {
        ReadableStream: Streams_1.streamConstructors.ReadableStream,
        WritableStream: Streams_1.streamConstructors.WritableStream,
        ...config,
        useDateObjects: (config.useDateObjects ?? false),
    };
    return new InstantReactNativeDatabase(configStrict, {
        '@instantdb/react': version_1.default,
    });
}
/**
 * @deprecated
 * `init_experimental` is deprecated. You can replace it with `init`.
 *
 * @example
 *
 * // Before
 * import { init_experimental } from "@instantdb/react-native"
 * const db = init_experimental({  ...  });
 *
 * // After
 * import { init } from "@instantdb/react-native"
 * const db = init({ ...  });
 */
const init_experimental = init;
exports.init_experimental = init_experimental;
class InstantReactNativeDatabase extends react_common_1.InstantReactAbstractDatabase {
    static Store = Storage_1.default;
    static NetworkListener = NetworkListener_1.default;
    static EventSourceImpl = EventSourceImpl_1.default;
}
exports.InstantReactNativeDatabase = InstantReactNativeDatabase;
