// @instantdb/react-native/dist/Storage.native.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
const async_storage_1 = __importDefault(require("@react-native-async-storage/async-storage"));
const core_1 = require("@instantdb/core");
const version = 5;
class Store extends core_1.StoreInterface {
    appId;
    dbName;
    constructor(appId, dbName) {
        super(appId, dbName);
        this.appId = appId;
        this.dbName = dbName;
    }
    makeKey(k) {
        return `instant_${this.appId}_${version}_${this.dbName}_${k}`;
    }
    async getItem(k) {
        const res = await async_storage_1.default.getItem(this.makeKey(k));
        if (res === null)
            return res;
        return JSON.parse(res);
    }
    async setItem(k, v) {
        await async_storage_1.default.setItem(this.makeKey(k), JSON.stringify(v));
    }
    async removeItem(k) {
        await async_storage_1.default.removeItem(this.makeKey(k));
    }
    async getAllKeys() {
        const allKeys = await async_storage_1.default.getAllKeys();
        const keys = [];
        const keyPrefix = this.makeKey('');
        for (const key of allKeys) {
            if (key.startsWith(keyPrefix)) {
                keys.push(key.substring(keyPrefix.length));
            }
        }
        return keys;
    }
    async multiSet(keyValuePairs) {
        await async_storage_1.default.multiSet(keyValuePairs.map(([k, v]) => [this.makeKey(k), JSON.stringify(v)]));
    }
}
exports.default = Store;
